lm_scorer/models/abc/base.py

```diff
@@ -11,31 +11,38 @@ def __init__(self, model_name: str, **kwargs: Any) -> None:
         self._build(model_name, kwargs)
 
     def sentence_score(
-        self, text: str, log: bool = False, reduce: str = "prod"
-    ) -> float:
-        log_probs, _, _ = self._tokens_log_prob(text)
-        tlen = log_probs.shape[0]
-
-        if reduce == "prod":
-            score = log_probs.sum()
-        elif reduce == "mean":
-            score = log_probs.logsumexp(0) - math.log(tlen)
-        elif reduce == "gmean":
-            score = log_probs.mean(0)
-        elif reduce == "hmean":
-            score = log_probs.neg().logsumexp(0).neg() + math.log(tlen)
-        else:
-            raise ValueError("Unrecognized scoring strategy: %s" % reduce)
+        self, sentences: Union[str, Iterable[str]], log: bool = False, reduce: str = "prod"
+    ) -> Union[float, List[float]]:
+        sentences = [sentences] if type(sentences) == str else sentences
+        outputs = self._tokens_log_prob(sentences)
 
-        if not log:
-            score = score.exp()
+        scores = []
+        for output in outputs:
+            log_probs = output[0]
+            tlen = log_probs.shape[0]
+
+            if reduce == "prod":
+                score = log_probs.sum()
+            elif reduce == "mean":
+                score = log_probs.logsumexp(0) - math.log(tlen)
+            elif reduce == "gmean":
+                score = log_probs.mean(0)
+            elif reduce == "hmean":
+                score = log_probs.neg().logsumexp(0).neg() + math.log(tlen)
+            else:
+                raise ValueError("Unrecognized scoring strategy: %s" % reduce)
+
+            if not log:
+                score = score.exp()
+
+            scores.append(score.item())
 
-        return score.item()
+        return scores[0] if len(scores) == 1 else scores
```

Comment:
```suggestion
        return scores[0] if type(text) == str else scores
```
Otherwise if the user inputs a list of strings with a single string then you return a string instead of a list of string. We want to give in output the same type as we receive in input.